packages/pluggableWidgets/tree-view-web/src/components/TreeView.tsx

```diff
@@ -130,14 +142,21 @@ const treeViewBodyClassName = "widget-tree-view-body";
 
 function TreeViewBranch(props: TreeViewBranchProps): ReactElement {
     const { level: currentContextLevel } = useContext(TreeViewBranchContext);
-    const [treeViewIsExpanded, setTreeViewIsExpanded] = useState<boolean>(props.startExpanded);
+    const [treeViewState, setTreeViewState] = useState<"COLLAPSED_WITH_JS" | "COLLAPSED_WITH_CSS" | "EXPANDED">(
+        !props.shouldLazyLoad && props.startExpanded ? "EXPANDED" : "COLLAPSED_WITH_JS"
+    );
     const [isActualLeafNode, setIsActualLeafNode] = useState<boolean>(props.isUserDefinedLeafNode || !props.children);
 
     const toggleTreeViewContent = useCallback(() => {
         if (!isActualLeafNode) {
-            setTreeViewIsExpanded(isExpanded => !isExpanded);
+            setTreeViewState(treeViewState => {
+                if (treeViewState === "COLLAPSED_WITH_JS" || treeViewState === "COLLAPSED_WITH_CSS") {
+                    return "EXPANDED";
+                }
+                return props.shouldLazyLoad ? "COLLAPSED_WITH_CSS" : "COLLAPSED_WITH_JS";
```

Comment:
Would be nice to extract into an enum.

Reply:
That's a good one, will do

Reply:
Had to rebase because #753 was merged and introduced merge conflicts. 6b34bc3 contains the enum change